hazelcast-client/src/main/java/com/hazelcast/client/connection/nio/ClientConnectionManagerImpl.java

```diff
@@ -561,14 +584,26 @@ public void run() {
     private void authenticated(Address target, ClientConnection connection) {
         ClientConnection oldConnection = connections.put(connection.getRemoteEndpoint(), connection);
         if (oldConnection == null) {
+            if (logger.isFinestEnabled()) {
+                logger.finest("Authentication succeeded for " + connection
+                        + " and there was no old connection to this end-point");
+            }
             fireConnectionAddedEvent(connection);
+        } else {
+            if (logger.isFinestEnabled()) {
+                logger.finest("Authentication succeeded for " + connection + " and replaced " + oldConnection
+                    + " Closing the old connection.");
+                destroyConnection(oldConnection, "Replaced by newer connection", null);
+            }
         }
         assert oldConnection == null || connection.equals(oldConnection);
         connectionsInProgress.remove(target);
     }
 
-    private void failed(Address target, ClientConnection connection, Throwable cause) {
-        logger.finest(cause);
+    private void authenticationFailed(Address target, ClientConnection connection, Throwable cause) {
```

Comment:
I would call this method 'onAuthenticationFailed' because it indicated that you want to do something as a consequence of a failed authentication.